### PR TITLE
fix: no route defaults are applied if 'protocols' is used

### PR DESCRIPTION
**Pull request by GGabriele**
Right now, Koko doesn't apply routes' entity schema defaults when 'protocols' is not empty. The reason for this was probably an attempt to make schema validation easier: the `strip_path` default value is `true`, which is not compatible with `grpc` and `grpcs` protocols; when `protocols` is not empty defaults are not applied to prevent a incompatibility like the one about `strip_path` to occur.
This also prevents other defaults to be applied, which is not desiderable.

This commit removes this restriction by adding a check on the content of the `protocols` field, setting `strip_path` to `false` in the case it's not provided as an input.

### CODE REVIEW

File: internal/resource/route.go [hunk 1]
```diff
@@ -126,13 +127,19 @@ func (r Route) ProcessDefaults(ctx context.Context) error {
...
 	}
-	if len(r.Route.Protocols) == 0 {
-		err := mergo.Merge(r.Route, defaultRoute,
-			mergo.WithTransformers(wrappersPBTransformer{}))
-		if err != nil {
-			return err
+	// When protocols has 'grpc' or 'grpcs', 'strip_path' cannot be set.
+	// If 'strip_path' is null, the default value (true) would be applied,
+	// so we set it to 'false' in order to prevent that.
+	if lo.Contains(r.Route.Protocols, "grpc") || lo.Contains(r.Route.Protocols, "grpcs") {
```
Comment (tjasko) [inline internal/resource/route.go:133]: There's a better way to do that w/ `lo` 😀
```suggestion
	if lo.Some(r.Route.Protocols, []string{typedefs.ProtocolGRPC, typedefs.ProtocolGRPCS}) {
```

Comment (tjasko) [inline internal/resource/route.go:133]: No test for this?

Comment (GGabriele) [reply internal/resource/route.go:133]: oh, nice!

Comment (GGabriele) [reply internal/resource/route.go:133]: It's already there: https://github.com/Kong/koko/blob/main/internal/resource/route_test.go#L901-L927

Comment (tjasko) [reply internal/resource/route.go]: Ah part of the validation test, that works w/ me.

**Review by tjasko — APPROVED** [review]

**Comment by hbagdi** [comment]
Ouch. This conditional defaulting is very painful and causes a lot of trouble. We need to get rid of it.
No action is required in this PR.

**Pull request merged**